### PR TITLE
fix: Do gracefull shutdown of Unleash on 'SIGINT' & 'SIGTERM'

### PR DESCRIPTION
**Pull request by ivarconr**
fixes: #706, #680

### CODE REVIEW

File: src/lib/types/core.ts [hunk 1]
```diff
@@ -18,6 +18,6 @@ export interface IUnleash {
     stores: IUnleashStores;
     server?: http.Server | https.Server;
     services: IUnleashServices;
-    stop: () => void;
+    stop: () => Promise<void>;
```
Comment (ivarconr) [inline src/lib/types/core.ts:21]: It's just stupid to not allow blocking until this has actually completed.

File: src/lib/app.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ export default function getApp(
     stores: IUnleashStores,
     services: IUnleashServices,
     eventBus?: EventEmitter,
-): any {
+): Application {
```
Comment (ivarconr) [inline src/lib/app.ts:33]: just had to fix the type.

File: websitev2/docs/deploy/configuring-unleash.md [hunk 3]
```diff
@@ -120,18 +120,44 @@ unleash.start(unleashOptions);
...
+
+### Graceful shutdown {#shutdown-unleash}
+
+> PS! Unleash will listen for the `SIGINT` signal and automatically trigger graceful shutdown of Unleash.
+
+If you need to stop Unleash (close database connections, and stop running Unleash tasks) you may use the stop function. Be aware that it is not possible to restart the Unleash instance after stopping it, but you can create a new instance of Unleash.
+
+```js
+const unleash = require('express');
+const unleash = require('unleash-server');
```
Comment (chriswk) [inline websitev2/docs/deploy/configuring-unleash.md:146]:
```suggestion
const express = require('express');
const unleash = require('unleash-server');
```

**Review by chriswk — APPROVED** [review]
Except the double usage of const unleash this LGTM :)

**Pull request merged**